src/main/java/graphql/Assert.java

```diff
@@ -10,6 +10,10 @@
         throw new AssertException(errorMessage);
     }
 
+    public static void assertNeverCalled() {
```

Comment:
if you make this 
    public static <T> T assertNeverCalled()

then you can do this at call site

   {
     return assertShouldNeverBecalled()
   }